panct/build_node_sample_map.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env sh
+
+# Extract walks (W lines) from a GFA file, flip them so that they can be queried by node instead of sample, and write the results to a TSV file
+
+# arg1: GFA file from which to extract walks
+# arg2 (optional): Path to .walk tab separated file to which to write output. Defaults to stdout
+
+# ex: ./build_node_sample_map.sh tests/data/basic.gfa basic.walk
+
+
+{ if [ "${1##*.}" = "gz" ]; then
+  gzip -dc "$1"
+else
+  cat "$1"
+fi } | grep -E '^W' | cut -f 2,7 | sed 's/	//;s/</	/g;s/>/	/g' | awk -F '	' -v OFS='	' '{
+  for (i = 1; i <= NF; i++) {
+    if (i == 1) {
+      key = $i;
+    } else {
+      print $i, key;
+    }
+  }
+}' | sort -u -t '	' -k1,1n -k2,2 | awk -F '	' -v OFS='	' '
```

Comment:
```suggestion
}' | sort -t '	' -k1,1n -k2,2 | awk -F '	' -v OFS='	' '
```
If a walk passes through a node more than once, the `-u` here will make it seem like the walk passed through this node only once. But I'm a bit concerned that we will need to know how many times a walk passes through a node in order to compute complexity for `--metrics sequniq-normwalk`. Should we remove the `-u`, then?

Reply:
ok, I just chatted with @WillardFord about this, and I think I will probably remove the `-u` for now, since it probably doesn't hurt to store extra info in the .walk file. We can always remove that extra info later when we query from the file

It would be good to chat with others about this, though